.github/workflows/integration-test-workflow.yml

```diff
@@ -34,12 +34,20 @@ jobs:
             path: debian-package_unpack
 
         - name: purge
-          run: sudo dpkg -P mosquitto tedge tedge_mapper
+          run: sudo dpkg -P mosquitto tedge tedge_mapper collectd-core collectd_mapper
           continue-on-error: true
 
         - name: install mosquitto
           run: sudo apt-get --assume-yes install mosquitto
 
+          #install collectd-core
+        - name: install collectd-core
+          run:  sudo apt-get --assume-yes install collectd-core
+
+          #replace the default config file with tedge custom config file
+        - name: configure collectd
+          run: sudo cp "./configuration/contrib/collectd/tedge-collectd.conf"  "/etc/collectd/collectd.conf"
```

Comment:
isn't it better to do this only as part of the collectd_mapper pysys test that validates the effect of this configuration?

Reply:
You are right, somehow these changes should not be part of this PR. This PR is only for fixing the bug. I updated code.